pandas/io/sql.py

```diff
@@ -1836,7 +1836,20 @@ def __init__(self, *args, **kwargs) -> None:
 
         # this will transform time(12,34,56,789) into '12:34:56.000789'
         # (this is what sqlalchemy does)
-        sqlite3.register_adapter(time, lambda _: _.strftime("%H:%M:%S.%f"))
+        def _adapt(dt):
+            if dt.tzinfo is None:
```

Comment:
Why does have tzinfo impact this if neither formatting captures %z?

Reply:
I was afraid that the hour attribute would mean a different thing (utc vs. locale hour). But it does not, so I could simplify the code.

I added a test parameter as this situation was not covered actually.
Finally, I changed for an f-string for even faster formatting.